marshmallow_recipe/bake.py

```diff
@@ -183,7 +183,9 @@ def pre_load(self, data: dict[str, Any]) -> Any:
 def _get_field_default(field: dataclasses.Field[_T]) -> Any:
     default_factory = field.default_factory
     if default_factory is not dataclasses.MISSING:  # type: ignore
-        raise ValueError(f"Default factory is not supported for {field}")
+        origin = typing_inspect.get_origin(field.type) or field.type
+        if origin in (list, List, dict, Dict) or dataclasses.is_dataclass(field.type):
+            raise ValueError(f"Default factory is not supported for {field}")
     return field.default
```

Comment:
1. This `default` value will still be `None` for such fields and will get into the metadata and **will not affect** field's `required` flag
2. The `default` is used in resulted `marshmallow` fields, factory must be used as well then for consistency

Reply:
Yeap. Return was lost during complex types fight =)